underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
What's the first version it was supported in? We could also drop support
for node 0.12.x since it's pretty ancient by this point

On Mon, May 2, 2016, 18:44 Brent Pendergraft <notifications@github.com>
wrote:

> In underscore.js
> <https://github.com/jashkenas/underscore/pull/2523#discussion_r61817716>:
>
> > @@ -1009,11 +1009,21 @@
> >
> >    // Convert an object into a list of `[key, value]` pairs.
> >    _.pairs = function(obj) {
> > -    var keys = _.keys(obj);
> > -    var length = keys.length;
> > -    var pairs = Array(length);
> > -    for (var i = 0; i < length; i++) {
> > -      pairs[i] = [keys[i], obj[keys[i]]];
> > +    var pairs, i, length;
> > +    if (_.isMap(obj)) {
> > +      var entries = obj.entries();
>
> Yeah, I agree that Array.from is a cleaner solution. Unfortunately it
> isn't supported in node 0.12.*, hence the failed tests.
>
> > typeof Map'function'> Array.fromundefined
>
> Not sure on the question of whether or not to even use underscore with
> Maps. I saw #2517
> <https://github.com/jashkenas/underscore/issues/2517#issuecomment-216385065>
> and decided to submit a PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/underscore/pull/2523/files/742033f3de979e46a99d3e9553088e5897d49952#r61817716>
>
-- 

Adam K (mobile)